### PR TITLE
Invitation model #8

### PR DESCRIPTION
**Pull request by TedGu**
### What this PR does (required):
- Invitation model

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- this model can be created in MongoDB(version:  v4.4.4 )

### CODE REVIEW

File: server/models/Invitation.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      type: String,
+      required: true,
+    },
+  ],
+  toUser: [
+    {
+      type: String,      
+      required: true,
+    },
+  ],
```
Comment (dproc96) [inline server/models/Invitation.js:15]: Can you change these to references to the users?

File: server/models/Invitation.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require("mongoose");
+
+const invitationSchema = new mongoose.Schema({
```
Comment (dproc96) [inline server/models/Invitation.js:3]: It looks like all of these properties are arrays, is there a specific use case for that? If not you can change that

File: server/routes/invitation.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const router = express.Router();
+
+const { approve, reject } = require("../controllers/invitation");
+
+router.route("/:id/approve").patch(approve);
+router.route("/:id/reject").patch(reject);
```
Comment (dproc96) [inline server/routes/invitation.js:7]: There's no protection on these routes and no verification that the user updating the request is the user who received it, can you add that logic?

**Review by dproc96 — COMMENTED** [review]
Had a couple of comments about implementation

**Review by sina-jamshidi — APPROVED** [review]
good to merge

**Pull request merged**